src/control/lib/hardware/pciutils/provider.go

```diff
@@ -0,0 +1,64 @@
+//
+// (C) Copyright 2024 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package pciutils
+
+import (
+	"context"
+
+	"github.com/daos-stack/daos/src/control/lib/hardware"
+	"github.com/pkg/errors"
+)
+
+type ctxKey string
+
+const (
+	AccessKey ctxKey = "pciutilsAccess"
+)
+
+type (
+	accessProvider interface {
+		PCIeCapsFromConfig(cfgBytes []byte, dev *hardware.PCIDevice) error
+		Cleanup()
+	}
+)
+
+func Init(parent context.Context) (context.Context, error) {
+	return context.WithValue(parent, AccessKey, &api{}), nil
+}
```

Comment:
Is there some way to avoid needing the caller to init/fini the `Context`? Especially considering cleanup doesn't do anything at the moment.

Reply:
reworking...